core/io/pwrite_spec.rb

```diff
@@ -28,16 +28,42 @@
       @file.pread(6, 0).should == "foobar"
     end
 
+    it "calls #to_s on the object to be written" do
+      data = mock("to_s")
+      data.should_receive(:to_s).and_return("foo")
+      @file.pwrite(data, 0)
+      @file.pread(3, 0).should == "foo"
+    end
+
+    it "calls #to_int on the offset" do
+      offset = mock("to_int")
+      offset.should_receive(:to_int).and_return(2)
+      @file.pwrite("foo", offset)
+      @file.pread(3, 2).should == "foo"
+    end
+
     it "raises IOError when file is not open in write mode" do
       File.open(@fname, "r") do |file|
-        -> { file.pwrite("foo", 1) }.should raise_error(IOError)
+        -> { file.pwrite("foo", 1) }.should raise_error(IOError, "not opened for writing")
       end
     end
 
     it "raises IOError when file is closed" do
       file = File.open(@fname, "w+")
       file.close
-      -> { file.pwrite("foo", 1) }.should raise_error(IOError)
+      -> { file.pwrite("foo", 1) }.should raise_error(IOError, "closed stream")
+    end
+
+    it "raises a NoMethodError if #to_s cannot be called on the object to be written" do
+      -> {
+        @file.pwrite(BasicObject.new, 0)
+      }.should raise_error(NoMethodError, /undefined method `to_s'/)
+    end
```

Comment:
**suggestion:** there is another case that makes sense to cover.

When `#to_s` returns non-String value - then it's ignored and object is still converted to String with `rb_any_to_s` CRuby function:

```c
VALUE
rb_any_to_s(VALUE obj)
{
    VALUE str;
    VALUE cname = rb_class_name(CLASS_OF(obj));

    str = rb_sprintf("#<%"PRIsVALUE":%p>", cname, (void*)obj);

    return str;
}
```

that prints object in the following way:
```
"#<Object:0x0000000106095428>"
```

Reply:
I'm not sure if this is something we need to test here, if we add it here it should also be added to `IO#print`, `IO#write`, `IO#syswrite` and all other `IO` methods I'm forgetting. But this code is unrelated to IO, so I guess this should be tested in `object/to_s_spec.rb` or something else that's more generic.